create-rust-app_cli/src/main.rs

```diff
@@ -6,181 +6,364 @@ mod qsync;
+            conflicts_with = "add new service",
+            hide = true,
+            help = "rust file(s) to read type information from",
+        )]
+        qsync_input_files: Option<Vec<PathBuf>>,
+
+        #[arg(
+            long="output",
+            name="output file",
+            value_name = "OUTPUT",
+            num_args=1,
+            value_hint = ValueHint::FilePath,
+            value_hint = ValueHint::DirPath,
+            conflicts_with = "add new service",
+            hide = true,
+            help = "file to write generated types to",
+        )]
+        qsync_output_file: Option<PathBuf>,
+
+        #[arg(
+            short = 'd',
+            name = "Debug",
+            help = "Dry-run, prints to stdout",
+            hide = true
+        )]
+        qsync_debug: bool,
+
+        #[arg(
+            long = "new-service",
+            name = "add new service",
+            help = "Add a model & service for backend. (beta)",
+            conflicts_with = "query-sync"
+        )]
+        add_new_service: bool,
+    },
 }
 
 /// CREATE RUST APP
 ///
 /// A MODERN WAY TO BOOTSTRAP A RUST+REACT APP IN A SINGLE COMMAND
 fn main() -> Result<()> {
-    let unknown_opts = UnknownOpt::from_args();
-
-    let mut current_dir: PathBuf = fs::get_current_working_directory()?;
+    let cli = Cli::parse();
 
     project::check_cli_version()?;
 
-    if unknown_opts.target.is_some() {
-        current_dir = PathBuf::from(unknown_opts.target.unwrap());
-
-        if current_dir.exists() {
-            logger::error(&format!(
-                "Cannot create a project: {:#?} already exists.",
-                &current_dir
-            ));
-            return Ok(());
+    // did user provide sub commands?
+    match cli.command {
+        Some(command) => {
+            // determine command based on the subcommand used
+            match command {
+                Commands::Create {
+                    cli_mode,
+                    name,
+                    database,
+                    backendframework,
+                    plugins,
+                } => create_project(cli_mode, name, database, backendframework, plugins)?,
+                Commands::Configure {
+                    query_sync,
+                    qsync_input_files,
+                    qsync_output_file,
+                    qsync_debug,
+                    add_new_service,
+                } => configure_project(
+                    query_sync,
+                    qsync_input_files,
+                    qsync_output_file,
+                    qsync_debug,
+                    add_new_service,
+                )?,
+            };
+        }
+        None => {
+            // base command on presence of Name arg
+            match cli.name {
+                Some(name) => create_project(false, name, None, None, None)?,
+                None => configure_project(false, None, None, false, false)?,
+            };
         }
```

Comment:
This is the nested match I was referring to, on second look it's not really that bad, I guess.